agent/http_oss.go

```diff
@@ -3,10 +3,37 @@
 package agent
 
 import (
+	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/hashicorp/consul/agent/structs"
 )
 
-func (s *HTTPServer) parseEntMeta(req *http.Request, entMeta *structs.EnterpriseMeta) {
+func (s *HTTPServer) parseEntMeta(req *http.Request, entMeta *structs.EnterpriseMeta) error {
+	if headerNS := req.Header.Get("X-Consul-Namespace"); headerNS != "" {
+		return BadRequestError{Reason: "Invalid header: \"X-Consul-Namespace\" - Namespaces is a Consul Enterprise feature"}
+	}
+	if queryNS := req.URL.Query().Get("ns"); queryNS != "" {
+		return BadRequestError{Reason: "Invalid query parameter: \"ns\" - Namespaces is a Consul Enterprise feature"}
+	}
+	return nil
+}
+
+func (s *HTTPServer) handleUnknownEnterpriseFields(err error) error {
```

Comment:
```suggestion
func rewordUnknownEnterpriseFieldsError(err error) error {
```

I am not 💯 about this function name, but the original one implies that the error is handled which is not the case. And the function can be standalone.

Reply:
Your are absolutely correct. I changed the name accordingly.